core/arch/arm/kernel/generic_core_bootcfg.c

```diff
@@ -149,6 +146,9 @@ static struct map_area bootcfg_memory_map[] = {
 	{
 	 .type = DEVICE1_TYPE,
 	 .pa = DEVICE1_BASE, .size = DEVICE1_SIZE,
+#ifdef DEVICE1_VA_BASE
```

Comment:
For example, on i.MX6UL, GIC base address is 0xA00000, if take this value as virtual base adderss, then it conflicts with TAs' address which is lower than 32M. This is why I introduce user defined virtual address for device.